core/src/main/java/org/mskcc/cbio/portal/servlet/QueryBuilder.java

```diff
@@ -400,27 +408,48 @@ else if (sampleIds != null && sampleIds.length()>0) {
 					geneticProfileMap.put(profileId, GeneticProfileUtil.getProfile(profileId, geneticProfileList));
 				}
 				
-				List<String> samplesList = new ArrayList<>(Arrays.asList(sampleIds.split(" ")));
+				List<String> samplesList = new ArrayList<>(Arrays.asList(sampleIdsStr.split(" ")));
 				studySampleMap.put(cancerStudyId,samplesList);
 				
 				if (sampleIdsKey == null)
 		        {
-		            sampleIdsKey = SampleSetUtil.shortenSampleIds(sampleIds);
+		            sampleIdsKey = SampleSetUtil.shortenSampleIds(sampleIdsStr);
 		        }
 
             // multiple studies OR single virtual study
             } else { 
 				if (dataTypePriority != null) {
-					AnnotatedSampleSets annotatedSampleSets = new AnnotatedSampleSets(sampleSetList, dataTypePriority);
-					SampleList defaultSampleSet = annotatedSampleSets.getDefaultSampleList();
-					if (defaultSampleSet == null) {
-						continue;
-					}
-					List<String> sampleList = defaultSampleSet.getSampleList();
-					if(inputStudySampleMap.get(cancerStudyId).size()>0){
-						sampleList.retainAll(inputStudySampleMap.get(cancerStudyId));
-					}
-					studySampleMap.put(cancerStudyId, sampleList);
+                    if (sampleSetId.equals("-1") && sampleIdsStr != null && sampleIdsStr.length() > 0) { //using user customized case list
```

Comment:
its under a for loop. `for (String cancerStudyId : inputStudySampleMap.keySet()) {...`